src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1071,38 +1034,162 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
             cleanUpBuildings(event);
 
             //  Spawn Citizens
-            if (townHall != null && citizens.size() < maxCitizens)
-            {
+            if (townHall != null && citizens.size() < maxCitizens) {
                 int respawnInterval = Configurations.citizenRespawnInterval * 20;
                 respawnInterval -= (60 * townHall.getBuildingLevel());
 
-                if (event.world.getWorldTime() % respawnInterval == 0)
-                {
+                if (event.world.getWorldTime() % respawnInterval == 0) {
                     spawnCitizen();
                 }
             }
+
+            if (!RAID_HAS_HAPPENED && !subscribers.isEmpty() && world != null && !world.isDaytime() && Configurations.doBarbariansSpawn)
```

Comment:
Okay, RAID_HAS_HAPPENED checks if a raid already happened for this colony (name please lower case).
Anyway, Can you please move this raid related methods in an extra method which will get called here.
Additionally I think it shouldn't happen every night, make it a Math.random configurable in the config (1 = everyday, 2 = every second , 3 third and so on (by chance).
By default id set it to 3 which seems like a good amount to recover.